src/app/campaign/form/campaign-form.component.ts

```diff
@@ -0,0 +1,92 @@
+import { Component, OnInit } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map, withLatestFrom } from 'rxjs/operators';
+import { TabService } from 'ngx-prx-styleguide';
+import { AdvertiserService } from '../service/advertiser.service';
+import { CampaignModel } from '../../shared/model/campaign.model';
+import { UserService } from '../../core/user/user.service';
+
+@Component({
+  template: `
+    <form *ngIf="campaign">
+      <prx-fancy-field
+        label="Owner"
+        [model]="campaign" name="accountId" [select]="accountOptions$ | async" required>
+      </prx-fancy-field>
+
+      <prx-fancy-field
+      label="Campaign Name"
+        textinput [model]="campaign" name="name"  required>
+      </prx-fancy-field>
+
+      <prx-fancy-field
+        label="Advertiser"
+        [model]="campaign" name="advertiserId" [select]="advertiserOptions$ | async" required>
+      </prx-fancy-field>
+
+      <prx-fancy-field
+        label="Campaign Status"
+        [model]="campaign" name="type" [select]="typeOptions" required>
+      </prx-fancy-field>
+
+      <prx-fancy-field
+        label="Campaign Type"
+        [model]="campaign" name="status" [select]="statusOptions" required>
+      </prx-fancy-field>
+
+      <prx-fancy-field
+        label="Campaign Rep"
+        textinput [model]="campaign" name="repName" required>
+      </prx-fancy-field>
+
+      <prx-fancy-field
+        label="Campaign Notes"
+        textarea [model]="campaign" name="notes">
+      </prx-fancy-field>
+
+    </form>
+  `
+})
+export class CampaignFormComponent implements OnInit {
+  advertiserOptions$: Observable<(string | number)[][]>;
+  accountOptions$: Observable<any[][]>;
+  campaign: CampaignModel;
+  typeOptions = [
```

Comment:
I'm cool with not enforcing object mutation on these select options by runtime crash, but I added `readonly` because that seemed like a more TypeScripty compromise.